### ISSUE TITLE
Plugin Common Options: improve validation

### ISSUE THREAD

**Issue by yaauie** [elastic/logstash#11947]
A number of the plugin common options (e.g., `add_field`) can be configured in such a way as to create runtime exceptions later:

~~~
input {
  stdin {
    add_field => {
      # `@metadata` needs to be a k/v map, setting it to a String will create issues later when other plugins use it.
      "@metadata" => "LOL"
      # `@timestamp` needs to be a time object or an ISO8601-compliant string
      "@timestamp" => "ERROR"
      # `tags` needs to be an array of distinct strings
      "tags" => "NOPE"
    }
  }
}
~~~

I believe we can catch these at configuration time and prevent the pipeline from starting up (8.0 breaking change) or at least warning the user in the logs (7.x)

**Comment by JakeFromTheDark** [comment]
Good idea, but don't disable these fields to be modified in inputs totally, please. Typically, we sign messages with the used input using **`input.*.add_field.@metadata.input`** field, which is then parsed in filters for different behaviour depending on the input received the message. So let's rather validate the assigned datatype to these internal fields, please.
